src/pages/index.js

```diff
@@ -0,0 +1,293 @@
+  imageElement.addEventListener("click", () => {
+    const modalImage = previewModal.querySelector(".modal__image");
+    const modalCaption = previewModal.querySelector(".modal__caption");
+
+    modalImage.src = data.link;
+    modalImage.alt = data.name;
+    modalCaption.textContent = data.name;
+
+    openModal(previewModal);
+  });
+}
+
+// Function for Like button handler
+function setUpLikeButton(likeButton, data) {
+  if (data.isLiked) likeButton.classList.add("card__like-btn_liked");
+
+  likeButton.addEventListener("click", () => {
+    const isLiked = likeButton.classList.contains("card__like-btn_liked");
+    api
+      .handleLikeStatus(data._id, isLiked)
+      .then((updatedData) =>
+        likeButton.classList.toggle("card__like-btn_liked", updatedData.isLiked)
+      )
+      .catch(console.error);
+  });
+}
+
+// Function for Delete modal logic
+function openDeleteModal(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+
+  const deleteButton = deleteForm.querySelector(".modal__submit-btn");
+  showLoading(deleteButton, false, "Deleting..", "Delete"); // Reset button text to "Delete"
+
+  // Define a new delete handler for the delete action
+  const newDeleteHandler = (evt) => {
+    evt.preventDefault();
+    showLoading(deleteButton, true, "Deleting...", "Delete"); // Show "Deleting..." while processing
+
+    api
+      .deleteCard(selectedCardId)
+      .then(() => {
+        selectedCard.remove(); // Remove the selected card from DOM upon successful deletion
+        closeModal(deleteModal);
+      })
+      .catch((error) => console.error("Failed to delete card:", error))
+      .finally(() => showLoading(deleteButton, false, "Deleting", "Delete")); // Reset button text to "Delete"
+  };
+
+  // Remove any previously attached handler from deleteForm if it exists
+  deleteForm.removeEventListener("submit", deleteForm._deleteHandler);
+
+  // Store the new handler in a property and attach it to the delete form
+  deleteForm._deleteHandler = newDeleteHandler;
+  deleteForm.addEventListener("submit", newDeleteHandler, { once: true });
+
+  // Open the delete modal
+  openModal(deleteModal);
+}
+
+// Profile edit form submission
+editProfileForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  const saveButton = editProfileForm.querySelector(".modal__submit-btn");
+  showLoading(saveButton, true, "Saving...", "Save");
+
+  const name = editProfileForm.querySelector("#profile-name-input").value;
+  const about = editProfileForm.querySelector(
+    "#profile-description-input"
+  ).value;
+
+  api
+    .editUserInfo({ name, about })
+    .then((userData) => {
+      document.querySelector(".profile__name").textContent = userData.name;
+      document.querySelector(".profile__description").textContent =
+        userData.about;
+      closeModal(editModal);
+    })
+    .finally(() => showLoading(saveButton, false, "Saving...", "Save"));
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **resetting** and **catching errors**


```js
// define a function for changing the button text. It accepts 4 params (the 2 last are optional with default texts)
export function renderLoading(isLoading, button, buttonText='Save', loadingText='Saving...') {
  if (isLoading) {
    button.textContent = loadingText
  } else {
    button.textContent = buttonText
  }
}

// define a universal function that accepts a request function, event and a default loading text 
function handleSubmit(request, evt, loadingText = 'Saving...') {
 // You need to prevent the default action in any submit handler
  evt.preventDefault();

  // the button is always available inside `event` as `submitter`
  const submitButton = evt.submitter;
  // fix the initial button text
  const initialText = submitButton.textContent;
  // change the button text before requesting
  renderLoading(true, submitButton, initialText, loadingText);
  // call the request function to be able to use the promise chain
  request()
    .then(() => {
      // any form should be reset after a successful response 
      // evt.target is the form in any submit handler		
      evt.target.reset();
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
    .catch(console.error)      

    // and in finally we need to stop loading
    .finally(() => {
      renderLoading(false, submitButton, initialText);
    });
}

```

Here is an example of handling the profile form submit:

```js
function handleProfileFormSubmit(evt) {
  // create a request function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally`
    return editProfile(nameInput.value, jobInput.value).then((userData) => {
      userName.textContent = userData.name;
      userJob.textContent = userData.about;
    });
  }
  // here we call handleSubmit passing the request and event (if you want a different loading text then you need to pass the 3rd param)
  handleSubmit(makeRequest, evt);
}

```

So, this way you can remove a lot of code duplicating. You will not need to search buttons, pass initial button texts and so on.

`handleSubmit` and `renderLoading` should be placed in `utils.js`, because they are utility functions